content/200-concepts/150-data-platform/02-data-proxy.mdx

```diff
@@ -155,19 +155,50 @@ You are now ready to try the Data Proxy. Go ahead and start your app!
 
 The Prisma Data Proxy does not support the preview features [Interactive Transactions](/concepts/components/prisma-client/transactions#interactive-transactions-in-preview) or [Metrics](/concepts/components/prisma-client/metrics) yet. To use the Prisma Data Proxy, you must first remove `interactiveTransactions` and `metrics` from the `previewFeatures` section of your `prisma.schema` file. When you run `prisma generate --data-proxy`, you will receive an error if you have either of these preview features enabled.
 
-### Providing the connection string to Prisma Migrate
+### Providing the connection string to Prisma Migrate and Introspection
 
 Currently, the Data Proxy can only be used to _query_ your database with [Prisma Client](/concepts/components/prisma-client).
 
-Migrations can be only be run on a direct (non-proxied) database connection. Assuming that your schema uses `env("DATABASE_URL")`, you'll need to override the `DATABASE_URL` environment variable before running `prisma migrate`. You can declare another environment variable, for example `MIGRATE_DATABASE_URL`, and use that to override the connection:
+Migrations or Introspection can only be run on a direct (non-proxied) database connection for now. This affects the following CLI commands:
+
+- `db pull`
+- `migrate deploy`
+- `migrate dev`
+- `migrate diff` (some parameters only)
+- `migrate reset`
+- `migrate resolve`
+- `migrate status`
+- `db push`
+- `db drop`
+- `db execute`
+- `studio`
+
+Assuming that your Prisma schema uses `env("DATABASE_URL")` as the `url` of the `datasource` block, you'll need to temporarily override the `DATABASE_URL` environment variable before running any of these commands. You can store the direct database connection string in another environment variable, for example `NONDATAPROXY_DATABASE_URL`, and use that to override the connection before you execute one of the affectes commands:
+
+```js
+NONDATAPROXY_DATABASE_URL="postgresql://..."
+```
```

Comment:
This should probably be changed to how we usually show how to define an env var (and maybe even link to an explanation if we have any).

Reply:
@zachtil can you work on this PR, please and thanks?!

Reply:
I don't know how we usually deal with this.

The "easiest" variant does not work on Windows for example unfortunately. There you have different commands depending on if you are in `Cmd` (`set FOO=bar`) or Powershell (`$env:FOO="bar"`).

Reply:
If `dotenv` works across the board (Win, macOS, Linux), then let's always point to `dotenv` and also indicate why we suggest to use that.
We already have a fairly good page that describes how that works and why it's a good solution. https://www.prisma.io/docs/guides/development-environment/environment-variables#how-does-prisma-use-environment-variables

Reply:
Actually, the other option is to provide tabs for this. But I have some qualms about tabs in general.

A **Window** tab would show how to set the variables on windows, a **Unix (macOS, Linux)** tab would show the the relevant command for that.

Yet another option is to just have a step that lists the options for both environments.

Set the `NONDATAPROXY_DATABASE_URL` environment variable.
- In Unix environments (macOS, Linux):
   ```
  NONDATAPROXY_DATABASE_URL="postgresql://..."
  ```
- In Windows CLI:
  ```
  set NONDATAPROXY_DATABASE_URL=postgresql://...
  ```
- In Windows PowerShell:
  ```
  $env:NONDATAPROXY_DATABASE_URL="postgresql://..."
  ```

Reply:
I think Linux standalone needs to be `export FOO="bar"` actually.